en/mavgen_c/message_signing_c.md

```diff
@@ -0,0 +1,149 @@
+# C Message Signing (mavgen)
+
+One of the key features of [MAVLink 2](../guide/mavlink_2.md) is support for [Message Signing](../guide/message_signing.md) (authentication).
+
+The C libraries generated by *mavgen* provide almost everything needed to support signing in your MAVLink system.
+You will need to add some code to:
+
+* Handle the [SETUP_SIGNING](../messages/common.md#SETUP_SIGNING) message.
+* Setup and teardown signing on a link.
+* Save and load the secret key and timestamp in persistent storage
+* Implement a callback to define which (if any) unsigned messages will be accepted.
+
+
+## Secret Key Management (SETUP_SIGNING)
+
+A secret key is 32 bytes of binary data that are used to create message signatures that can be verified by other holders of the key.
+The general requirements for creating, storing, logging and sharing keys are covered in: [Message Signing > Secret Key Management](../guide/message_signing.md#secret_key).
+
```

Comment:
What we need here is actual instructions for using _this_ library. Suggest:
1. Code example for creating secret key from passphase
2. Code example for parsing SETUP_SIGNING message and getting key
3. Code example for saving key (maybe on linux/nuttx)
4. Code example for passing new key into library.

Most important is last one - how do we set the secret key to use in the library?

Reply:
I see that "Enabling Signing on a Channel" shows how to access the key.